bolt-modules/dir/lib/puppet/functions/dir/entries.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+# Enumerate files in a given directory
+Puppet::Functions.create_function(:'dir::entries', Puppet::Functions::InternalFunction) do
+  # @param dirname Absolute path or Puppet file path.
+  # @return Array of files in the given directory
+  dispatch :entries do
+    scope_param
+    required_param 'String', :dirname
+    return_type 'Array'
+  end
+
+  def entries(scope, dirname)
+    # Send Analytics Report
+    Puppet.lookup(:bolt_executor) {}&.report_function_call(self.class.name)
+    found = Puppet::Parser::Files.find_file(dirname, scope.compiler.environment)
+    unless found && Puppet::FileSystem.exist?(found)
+      raise Puppet::ParseErrorWithIssue.from_issue_and_stack(
+        Puppet::Pops::Issues::NO_SUCH_FILE_OR_DIRECTORY, file: filename
+      )
+    end
+
+    # Sort for testability
+    Dir.entries(dirname).sort
```

Comment:
Maybe `Dir.children` instead? I'm not sure that having `.` and `..` is particularly useful.